presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/JdbcRecordCursor.java

```diff
@@ -71,7 +49,36 @@
 
     public JdbcRecordCursor(JdbcClient jdbcClient, JdbcSplit split, List<JdbcColumnHandle> columnHandles)
     {
-        this.columnHandles = ImmutableList.copyOf(requireNonNull(columnHandles, "columnHandles is null"));
+        this.columnHandles = requireNonNull(columnHandles, "columnHandles is null").stream()
+                .toArray(JdbcColumnHandle[]::new);
+
+        booleanReadFunctions = new BooleanReadFunction[this.columnHandles.length];
```

Comment:
I'd make this `columnHandles.size()`

Reply:
That's being paranoid, but i didn't make a surely-immutable copy of `columnHandles`..